### PR TITLE
feat: delete default view for workspace

### PR DESCRIPTION
**Pull request by speed2exe**
- Add http api for removing default view id for a workspace

### CODE REVIEW

File: src/api/workspace.rs [hunk 1]
```diff
@@ -1158,6 +1159,25 @@ async fn put_workspace_default_published_view_handler(
...
 
+async fn delete_workspace_default_published_view_handler(
+  user_uuid: UserUuid,
+  workspace_id: web::Path<Uuid>,
+  state: Data<AppState>,
+) -> Result<Json<AppResponse<()>>> {
+  let uid = state.user_cache.get_user_uid(&user_uuid).await?;
+  state
+    .workspace_access_control
+    .enforce_action(&uid, &workspace_id.to_string(), Action::Write)
```
Comment (khorshuheng) [inline src/api/workspace.rs:1170]: If only workspace owner should be allowed to delete default published view, you can use enfore_role instead.

File: src/biz/workspace/publish.rs [hunk 2]
```diff
@@ -113,6 +113,16 @@ pub async fn set_workspace_default_publish_view(
   Ok(())
 }
 
+pub async fn unset_workspace_default_publish_view(
+  pg_pool: &PgPool,
+  user_uuid: &Uuid,
+  workspace_id: &Uuid,
+) -> Result<(), AppError> {
+  check_workspace_owner(pg_pool, user_uuid, workspace_id).await?;
```
Comment (khorshuheng) [inline src/biz/workspace/publish.rs:121]: If we use enforce role: Owner, we don't need to check workspace owner again.

File: src/api/workspace.rs (hunk 1 above)
Comment (speed2exe) [reply src/api/workspace.rs]: done

File: src/biz/workspace/publish.rs (hunk 2 above)
Comment (speed2exe) [reply src/biz/workspace/publish.rs]: Removed. In addition, I have remove 'check_workspace_owner' and use `enforce_role` on the caller instead for other operation that requires `check_workspace_owner`

**Review by khorshuheng — APPROVED** [review]
lgtm

**Pull request merged**